src/System Application/App/AppSource Gallery/src/AppSourceProductManager.Codeunit.al

```diff
@@ -0,0 +1,604 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+namespace System.Apps.AppSource;
+
+using System.Environment.Configuration;
+using System.Globalization;
+using System.Azure.Identity;
+using System.Utilities;
+using System.Environment;
+using System.Azure.KeyVault;
+using System.RestClient;
+
+/// <summary>
+/// Library for managing AppSource product retrival and usage.
+/// </summary>
+codeunit 2515 "AppSource Product Manager" implements "IAppSource Product Manager Dependencies"
+{
+    Access = Internal;
+    InherentEntitlements = X;
+    InherentPermissions = X;
+
+    #region Dependency Interface implementation
+    procedure AzureADTenant_GetCountryLetterCode(): Text[2]
```

Comment:
Why the addition of AzureADTenant/AzureKeyVault at the front of the function names?

Reply:
Next PR